### DISCUSSION TITLE
[Enhancement] Have an option to build UIs with blazor syntax instead of XAML

### DISCUSSION THREAD

**Discussion by smartprogrammer93** [dotnet/maui#92] [General] [answered]
Like the Mobile Blazor Bindings but support it natively so we dont have to do the bindings. Imagine needing to do the bindings for every control that is out there. Yet i understand how hard it can be to move all the work from XAML in Xamarin Forms. I have used Xamarin forms to build apps. and let me tell you, building UI with xamarin is way harder than building UI with Blazor. Blazor will do the same functionality using way fewer lines of code than needed for XAML. I think XAML should be the secondary citizen and not Blazor as a standard UI for MAUI.

**Comment by smartprogrammer93** [comment]
might be a Duplicate of #40

**Comment by GiampaoloGabba** [comment]
one big problem of XAML is the verbosity... i really hope that they can simplify it in MAUI, tho i will switch to MVU and UI in c# as soon as the first MAUI preview goes out :)

Btw i don't agree that XAML should be secondary citizen but i do agree that blazor syntax need more love and an architectural semplification, in this current state is not much usable...

**Comment by charlesroddie** [comment]
This is OK but markup languages should not be in the core of MAUI. The core is .Net objects. Markup languages describing these objects should be in separate layers and separate repos to improve maintainability of the core. The fewer markup languages maintained in the core the better: ideally 0, but 2 is worse than 1.

**Comment by smartprogrammer93** [comment]
@charlesroddie I don't disagree. but the core should be more acceptable of different Markup langauges. the current state of Xamarin Form does not make easy to use razor syntax as the markup langauge. I don't see a future for Mobile Blazor bindings for this reason. I myself was so excited once i saw its announcement. but once i saw how it works, I knew I will not be using it because of it being unreliable although I started using Blazor in my solutions since it was experimental. That is how much i love it.

**Comment by PureWeen** [comment, marked as answer]
> This is OK but markup languages should not be in the core of MAUI. The core is .Net objects. Markup languages describing these objects should be in separate layers and separate repos to improve maintainability of the core. The fewer markup languages maintained in the core the better: ideally 0, but 2 is worse than 1.

They aren't going to be part of maui core

https://github.com/dotnet/maui/issues/28

We'll have some separate dll like "Maui.Xaml" or something that consumes Maui.Core

> I don't disagree. but the core should be more acceptable of different Markup langauges. 

That's one of the biggest use cases for the Maui rework
